### PR TITLE
Adds a buckle check for entering mechs, works on all mobs

### LINKED ISSUE #11828: You can enter a mech while buckled to a chair
If you do so, you remain buckled to the chair and cannot move unless you resist to unbuckle.

### PR DESCRIPTION
**Pull request by Lijah99**
Fixes #11828 

This addresses tactical chairs 
I will then see if you can enter while restrained, if so, I will look into addressing that issue as well.

### CODE REVIEW

**Comment by Jordie0608** [comment]
If you say 'Fixes #11828' the issue will automatically be closed when this is merged.

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -664,6 +664,10 @@
 		user << "<span class='warning'>Access denied.</span>"
 		log_append_to_last("Permission denied.")
 		return
+	if(user)(mob_buckled)
```
Comment (RemieRichards) [inline code/game/mecha/mecha.dm]: ```if(user)(mob_buckled)``` --> ```if(user.buckled)```

Comment (Lijah99) [inline code/game/mecha/mecha.dm]: rip, Would this create an error, even if it works for both the player and if someone tried to do this with a mob?

**Comment by phil235** [comment]
You also need to add another check after the "do_after()", otherwise the bug could still happen if the mob trying to enter is buckled mid-action.

**Comment by Lijah99** [comment]
I need an adult

File: code/game/mecha/mecha.dm (hunk 1 above)
Comment (RemieRichards) [inline code/game/mecha/mecha.dm]: what you've written already isn't valid syntax (and byond actually complained in the Travis report)

**Comment by Lijah99** [comment]
There
;_;

**Pull request merged**